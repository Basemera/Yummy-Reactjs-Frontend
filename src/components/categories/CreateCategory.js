import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import axiosInstance from '../Apicalls';

class CreateCategory extends Component {
    state = {
        title: ''
    }
    handleInputChange = (event) => {
        const {name, value} = event.target;
        this.setState({[name]: value});
    }
    handleAddCategories = (event) => {
        const {title: name} = this.state;
        const headers = {
            Authorization: `Bearer ${localStorage.getItem('accessToken')}`
        }

        event.preventDefault();
        axiosInstance.post('categories/', {name}, {headers})
        .then(response => {
            this
                .props
                .getCategories()

        }).catch(error => {
            if (error.response) {
                alert(error.response.data.message)
            } else if (error.request) {
                alert("Request not made")
            }
        });
    }

    render() {
        return (
            <div >
                <div id="recipebar">
                    <div className="container text-center">
                        <div className='row'>
                            <div
                                className='col-xs-8  col-sm-6 offset-xs-2 offset-sm-3 col-md-4 offset-md-4'>
                                <h2>Create Your Recipe Category</h2>

                                <form onSubmit={this.handleAddCategories}>
                                    <div className="form-group">
                                        <input
                                            type="text"
                                            name="title"
                                            className='form-control'
                                            placeholder="category name"
                                            onChange={this.handleInputChange}
                                            value={this.state.title}/>
                                    </div>

                                    <input type="submit" className='btn btn-primary' value="Add Category"/>
                                </form>
                            </div>
                        </div>
                    </div>
                    <br/>
                    
                    <form className="form-group" role="search">
                        <div className="input-group add-on col-sm-3 pull-right">
                            <input type="text" className="form-control" placeholder="Search for a category..." name="srch-term"/>
                            <div className="input-group-btn">
                                <button className="btn btn-default" type="submit">
                                    <i className="glyphicon glyphicon-search"></i></button>
                            </div>
                        </div>
                    </form>
                    
                </div>
            </div>
        );
    }
}
export default CreateCategory;