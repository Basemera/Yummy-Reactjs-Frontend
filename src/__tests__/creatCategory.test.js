import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import CreateCategory from '../components/categories/CreateCategory';

describe('CreateCategory component', () => {
  const wrapper = shallow(<CreateCategory />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has initial state', () => {
    expect(wrapper.state().title).toEqual('');
  });

  it('changes state', () => {
    wrapper.setState({ title: 'lunch' });
    expect(wrapper.find('[name="title"]').props().value).toEqual('lunch');
  });

  it('has the correct form fields', () => {
    expect(wrapper.find('[name="title"]')).toHaveLength(1);
  });

  it('renders component in divs', () => {
    expect(wrapper.find('div')).toHaveLength(6);
  });

  it('renders component in spans', () => {
    expect(wrapper.find('span')).toHaveLength(1);
  });

  it('renders component in h2', () => {
    expect(wrapper.find('h2')).toHaveLength(1);
  });

  it('has input field', () => {
    expect(wrapper.find('input')).toHaveLength(2);
  });

  it('renders a form', () => {
    expect(wrapper.find('form')).toHaveLength(1);
  });
});
