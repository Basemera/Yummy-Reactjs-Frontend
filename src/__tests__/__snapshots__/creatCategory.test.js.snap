// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component renders properly 1`] = `
<div>
  <div
    id="recipebar"
  >
    <div
      className="container text-center word-font"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-8  col-sm-6 offset-xs-2 offset-sm-3 col-md-4 offset-md-4"
        >
          <h2
            className="word-font"
          >
            Create a category
          </h2>
          <form
            onSubmit={[Function]}
          >
            <div
              className="input-group "
            >
              <span
                class="input-group-addon"
              >
                <i
                  class="glyphicon glyphicon-pencil"
                />
              </span>
              <input
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="category name"
                type="text"
                value=""
              />
            </div>
            <br />
            <input
              className="btn btn-primary"
              type="submit"
              value="Add Category"
            />
          </form>
        </div>
      </div>
    </div>
    <br />
  </div>
</div>
`;
